referral_module/api/views.py

```diff
@@ -0,0 +1,55 @@
+from .serializers import CampaignSerializer, CampaignDetailsSerializer, CampaignUserReferralSerializer, UserSerializer
+from rest_framework import generics, permissions
+from referral_module import models
+from django.contrib.auth.models import User
+
+
+class UserList(generics.ListAPIView):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+
+class UserDetail(generics.RetrieveAPIView):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+
+class CampaignList(generics.ListAPIView):
+    queryset = models.Campaign.objects.all()
+    serializer_class = CampaignSerializer
+
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+
+
+class CampaignDetail(generics.RetrieveAPIView):
+    queryset = models.Campaign.objects.all()
+    serializer_class = CampaignDetailsSerializer
+
+    permission_classes = (permissions.IsAuthenticated,)
+
+
+class CampaignReferralNumber(
+    generics.CreateAPIView,
+    generics.RetrieveAPIView
+):
+
+    queryset = models.Campaign.objects.all()
```

Comment:
I think it shouldn't be possible to request the key for the campaign which is not active.